src/java/fr/paris/lutece/plugins/forms/web/admin/FormResponseJspBean.java

```diff
@@ -391,19 +403,109 @@ public String doFormResponseSummary( HttpServletRequest request )
     public synchronized String doSaveFormResponseSummary( HttpServletRequest request )
     {
         Form form = null;
-        
-        try 
+
+        try
         {
-			form = findFormFrom( request );
-		} 
+            form = findFormFrom( request );
+        }
         catch (FormNotFoundException e1)
         {
-			return getStepView( request );
-		}
-        
-    	return doSaveResponse(request, form);
+            return getStepView( request );
+        }
+
+        return doSaveResponse(request, form);
+    }
+
+    /**
+     * Adds an iteration
+     *
+     * @param request
+     *            the request
+     * @return the XPage
+     * @throws SiteMessageException
+     *             if there is an error during the iteration
+     * @throws UserNotSignedException
+     *             if the user is not signed in
+     */
+    @Action( value = ACTION_ADD_ITERATION )
+    public synchronized String doAddIteration( HttpServletRequest request ) throws SiteMessageException, UserNotSignedException
+    {
+        IsRequestComingFromAction = true;
+
+        try
+        {
+            boolean bSessionLost = isSessionLost( );
+            findFormFrom( request );
+            if ( bSessionLost )
+            {
+                addWarning( MESSAGE_WARNING_LOST_SESSION, getLocale() );
+                return getStepView(  request );
+            }
+
+            FormsResponseUtils.fillResponseManagerWithResponses( request, false, _formResponseManager, _stepDisplayTree.getQuestions( ), false );
+        }
+        catch( FormNotFoundException | QuestionValidationException exception )
+        {
+            return getStepView(  request );
+        }
+
+        int nIdGroup = NumberUtils.toInt( request.getParameter( FormsConstants.PARAMETER_ACTION_PREFIX + ACTION_ADD_ITERATION ),
+                FormsConstants.DEFAULT_ID_VALUE );
+
+        if ( nIdGroup != FormsConstants.DEFAULT_ID_VALUE )
+        {
+            _stepDisplayTree.iterate( nIdGroup );
+        }
+
+        return getStepView(  request );
     }
-    
+
+    /**
+     * Remove an iteration
+     *
+     * @param request
+     *            the request
+     * @return the XPage
+     * @throws SiteMessageException
+     *             if there is an error during the iteration
+     * @throws UserNotSignedException
+     *             if the user is not signed in
+     */
+    @Action( value = ACTION_REMOVE_ITERATION )
+    public synchronized String doRemoveIteration( HttpServletRequest request ) throws SiteMessageException, UserNotSignedException
```

Comment:
SiteMessageException, UserNotSignedException !!! Why throw these two exceptions? There will never be an exception of this type in this code.

Reply:
So I deleted the throw on this exceptions